### PR TITLE
№ 2544 » BE | Move 'book open days title' from ProgrammePageGlobalFieldsSettings to ProgrammePage

### PR DESCRIPTION
**Pull request by engineervix**
[Ticket](https://projects.torchbox.com/projects/rca-django-cms-project/tickets/2544)

This PR moves the 'book open days title' setting from `ProgrammePageGlobalFieldsSettings` to `ProgrammePage`, so that editors can customize this for each programme page, rather than globally for all programme pages. 

<details><summary>Screenshots</summary>

[Screenshot - Wagtail Admin](https://github.com/torchbox/rca-wagtail-2019/assets/7713776/cb7c37d7-f19e-45e1-8436-0ef85c37a26f "Screenshot - Wagtail Admin")

[Screenshot - Template](https://github.com/torchbox/rca-wagtail-2019/assets/7713776/5fe3180a-f121-401d-bf8c-8e9f4cdbc31b "Screenshot - Template")

</details>

### CODE REVIEW

File: rca/programmes/migrations/0089_remove_programmepageglobalfieldssettings_key_details_book_or_view_all_open_days_link_title_and_more.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+            model_name="programmepage",
+            name="book_or_view_all_open_days_link_title",
+            field=models.CharField(
+                blank=True,
+                default="Book or view all open days",
+                max_length=255,
+                verbose_name="Book open days title",
+            ),
+        ),
+    ]
```
Comment (kevinhowbrook) [inline rca/programmes/migrations/0089_remove_programmepageglobalfieldssettings_key_details_book_or_view_all_open_days_link_title_and_more.py:27]: **Suggestion**

Rename the migration file to something simpler.

File: rca/programmes/models.py [hunk 2]
```diff
@@ -380,6 +380,12 @@ class ProgrammePage(TapMixin, ContactFieldsMixin, BasePage):
 
     next_open_day_date = models.DateField(blank=True, null=True)
     link_to_open_days = models.URLField(blank=True)
+    book_or_view_all_open_days_link_title = models.CharField(
+        max_length=255,
+        blank=True,
```
Comment (kevinhowbrook) [inline rca/programmes/models.py:385]: I think it's worth adding validation for this to be required if the link_to_open_days is set. At the moment you could remove the default text and have a blank link.

**Review by kevinhowbrook — APPROVED** [review]

File: rca/programmes/models.py (hunk 2 above)
Comment (engineervix) [reply rca/programmes/models.py:385]: Thanks! Added in 03b44ba6

<details><summary>Screenshot</summary>

[Screenshot](https://github.com/torchbox/rca-wagtail-2019/assets/7713776/7a97a2fe-e619-4374-80c1-bad24e691493)


</details>

File: rca/programmes/migrations/0089_remove_programmepageglobalfieldssettings_key_details_book_or_view_all_open_days_link_title_and_more.py (hunk 1 above)
Comment (engineervix) [reply rca/programmes/migrations/0089_remove_programmepageglobalfieldssettings_key_details_book_or_view_all_open_days_link_title_and_more.py]: Thanks, done in 37712d16

**Pull request merged**